main/src/com/google/refine/expr/functions/ToString.java

```diff
@@ -54,11 +57,14 @@ public Object call(Properties bindings, Object[] args) {
                     OffsetDateTime odt = (OffsetDateTime)o1;
                     return odt.format(DateTimeFormatter.ofPattern((String)o2));
                 } else if (o1 instanceof Number) {
-                    return String.format((String) o2, (Number) o1);
+                    DecimalFormat df = new DecimalFormat((String) o2);
+                    return df.format(o1);
                 }
             } else if (args.length == 1) {
                 if (o1 instanceof String) {
                     return (String) o1;
+                } else if (o1 instanceof Number) {
+                    return NumberFormat.getInstance().format(o1);
```

Comment:
The "default" transformations to strings for other objects are in the com.google.refine.util.StringUtils class - wouldn't it make sense for this default number->string behaviour to also be in that class also?

Reply:
Note that there is already some default behaviour for number.toString() in GREL (I haven't tracked down where this happens - but it definitely already works - do we need any additional code here?)

Reply:
This is true @ostephens 
I talked this out with Antonin and resolved that this may have already been fixed. I pushed some commits showing tests which demonstrate that and a try-catch for invalid formats.